### PR TITLE
[FLOC-1266] Release 0.3.3dev4

### PR DESCRIPTION
**Pull request by adamtheturtle**
This branch should not be merged until the release process is complete.

https://clusterhq.atlassian.net/browse/FLOC-1266

### CODE REVIEW

**Comment by adamtheturtle** [comment]
The build is still in progress but I guess that the files changed can be checked already.

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+
+- Applications can now be configured with a limit on the maximum size of their volume. (FLOC-729)
+- Applications can now be configured with an automatic restart policy. (FLOC-727)
+- Docker 1.2 is now required. (FLOC-721)
+- The documentation has a new theme. (FLOC-1257)
+- Omnibus packages are now used for distribution. (FLOC-508)
+- Added instructions for reporting security issues. (FLOC-1183)
+- Added a Volume API version endpoint. (FLOC-1162)
+- Flocker no longer conflicts with Homebrew's Python. (FLOC-660)
+- Flocker is now supported on Fedora 20 on Rackspace Cloud Servers (FLOC-965)
```
Comment (wallrj) [inline NEWS]: Also on AWS: http://doc-dev.clusterhq.com/gettinginvolved/acceptance-testing.html#aws

File: NEWS [hunk 2]
```diff
@@ -1,3 +1,16 @@
+Flocker 0.3.3dev4 (2015-01-27)
+==============================
+
+- Applications can now be configured with a limit on the maximum size of their volume. (FLOC-729)
+- Applications can now be configured with an automatic restart policy. (FLOC-727)
+- Docker 1.2 is now required. (FLOC-721)
```
Comment (wallrj) [inline NEWS]: Actually, the minimum seems to be 1.3. 
* https://github.com/ClusterHQ/flocker/blame/master/admin/packaging.py#L192

It's not as easy to see, at a glance, the package dependencies with the omnibus packaging system.
Perhaps we should try and improve that.
I think the version got incremented when we started using a newer version of the Docker API.

File: LICENSE [hunk 3]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2014 ClusterHQ
+   Copyright 2014, 2015 ClusterHQ
```
Comment (wallrj) [inline LICENSE]: Are we now ClusterHQ Inc?

Comment (adamtheturtle) [inline LICENSE]: I'll leave this - we can change the company name everywhere at once (at the top of source files for example) when necessary.

**Comment by wallrj** [comment]
There's one suspicious broken link:

```
gettinginvolved/contributing.rst:44: [broken] https://github.com/ClusterHQ/flocker/compare/release/flocker-1.2.3...release-maintenance/flocker-1.2.3/fix-a-bug-FLOC-1234?expand=1: HTTP Error 404: Not Found
```
http://build.clusterhq.com/builders/flocker-docs/builds/3818/steps/build-linkcheck/logs/errors

But that can be fixed in another issue. And one solution would be for it to be plain text rather than a link.

**Comment by wallrj** [comment]
Seems that you have to specify the `--flocker-version` when running the acceptance tests. Docs need updating:

``admin/run-acceptance-tests --distribution fedora-20 --flocker-version=0.3.2.1225.g742fd6d``

Where version is the version shown by `vagrant box list`

```
(flocker-release-0.3.3dev4)(release/flocker-0.3.3dev4 =)[~/.../HybridLogic/flocker-0.3.3dev4]$ vagrant box list
centos65-x86_64-20140116   (virtualbox, 0)
chef/centos-7.0            (virtualbox, 1.0.0)
clusterhq/fedora20-updated (virtualbox, 2014.09.19)
clusterhq/flocker-dev      (virtualbox, 0.3.2.324.g7987f9b)
clusterhq/flocker-tutorial (virtualbox, 0.3.2.1225.g742fd6d)
ubuntu/trusty64            (virtualbox, 14.04)
```

**Comment by wallrj** [comment]
I'm seeing some errors when running acceptance tests locally eg

```
(flocker-release-0.3.3dev4)(release/flocker-0.3.3dev4 =)[~/.../HybridLogic/flocker-0.3.3dev4]$ admin/run-acceptance-tests --distribution fedora-20 --flocker-version=0.3.2.1225.g742fd6d
==> node2: VM not created. Moving on...
==> node1: VM not created. Moving on...
Bringing machine 'node1' up with 'virtualbox' provider...
...

[root@172.16.255.241] out: Status: Image is up to date for clusterhq/mongodb:latest
[root@172.16.255.241] out: 

Disconnecting from root@172.16.255.241... done.
flocker.acceptance.test_deployment
  DeploymentTests
    test_application_volume_quotas ... Unhandled Error
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 784, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib64/python2.7/threading.py", line 811, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 764, in run
    self.__target(*self.__args, **self.__kwargs)
--- <exception caught here> ---
  File "/opt/flocker/lib/python2.7/site-packages/twisted/python/threadpool.py", line 196, in _worker
    result = context.call(ctx, function, *args, **kwargs)
  File "/opt/flocker/lib/python2.7/site-packages/twisted/python/context.py", line 118, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/opt/flocker/lib/python2.7/site-packages/twisted/python/context.py", line 81, in callWithContext
    return func(*args,**kw)
  File "/opt/flocker/lib/python2.7/site-packages/flocker/node/_docker.py", line 432, in _add
    _create()
  File "/opt/flocker/lib/python2.7/site-packages/flocker/node/_docker.py", line 427, in _create
    host_config=host_config,
  File "/opt/flocker/lib/python2.7/site-packages/docker/client.py", line 556, in create_container
    return self.create_container_from_config(config, name)
  File "/opt/flocker/lib/python2.7/site-packages/docker/client.py", line 564, in create_container_from_config
    return self._result(res, True)
  File "/opt/flocker/lib/python2.7/site-packages/docker/client.py", line 98, in _result
    self._raise_for_status(response)
  File "/opt/flocker/lib/python2.7/site-packages/docker/client.py", line 94, in _raise_for_status
    raise errors.APIError(e, response, explanation=explanation)
docker.errors.APIError: 500 Server Error: Internal Server Error ("Could not create local directory '/flocker/899fd0d6-1899-4ad3-967a-a0a6228af1b8.default.565d4aee-9a88-43c4-ad5c-32ba0ea2941d' for bind mount: mkdir /flocker/899fd0d6-1899-4ad3-967a-a0a6228af1b8.default.565d4aee-9a88-43c4-ad5c-32ba0ea2941d: read-only file system!")
main function encountered error
Traceback (most recent call last):
Failure: twisted.internet.defer.FirstError: FirstError[#0, [Failure instance: Traceback: <class 'docker.errors.APIError'>: 500 Server Error: Internal Server Error ("Could not create local directory '/flocker/899fd0d6-1899-4ad3-967a-a0a6228af1b8.default.565d4aee-9a88-43c4-ad5c-32ba0ea2941d' for bind mount: mkdir /flocker/899fd0d6-1899-4ad3-967a-a0a6228af1b8.default.565d4aee-9a88-43c4-ad5c-32ba0ea2941d: read-only file system!")
/usr/lib64/python2.7/threading.py:784:__bootstrap
/usr/lib64/python2.7/threading.py:811:__bootstrap_inner
/usr/lib64/python2.7/threading.py:764:run
--- <exception caught here> ---
/opt/flocker/lib/python2.7/site-packages/twisted/python/threadpool.py:196:_worker
/opt/flocker/lib/python2.7/site-packages/twisted/python/context.py:118:callWithContext
/opt/flocker/lib/python2.7/site-packages/twisted/python/context.py:81:callWithContext
/opt/flocker/lib/python2.7/site-packages/flocker/node/_docker.py:432:_add
/opt/flocker/lib/python2.7/site-packages/flocker/node/_docker.py:427:_create
/opt/flocker/lib/python2.7/site-packages/docker/client.py:556:create_container
/opt/flocker/lib/python2.7/site-packages/docker/client.py:564:create_container_from_config
/opt/flocker/lib/python2.7/site-packages/docker/client.py:98:_result
/opt/flocker/lib/python2.7/site-packages/docker/client.py:94:_raise_for_status
]]
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 784, in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib64/python2.7/threading.py", line 811, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 764, in run
    self.__target(*self.__args, **self.__kwargs)
--- <exception caught here> ---
  File "/home/richard/.virtualenvs/flocker-release-0.3.3dev4/lib/python2.7/site-packages/twisted/python/threadpool.py", line 196, in _worker
    result = context.call(ctx, function, *args, **kwargs)
  File "/home/richard/.virtualenvs/flocker-release-0.3.3dev4/lib/python2.7/site-packages/twisted/python/context.py", line 118, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/home/richard/.virtualenvs/flocker-release-0.3.3dev4/lib/python2.7/site-packages/twisted/python/context.py", line 81, in callWithContext
    return func(*args,**kw)
  File "/home/richard/projects/HybridLogic/flocker-0.3.3dev4/flocker/common/_ipc.py", line 90, in get_output
    raise IOError("Bad exit", remote_command, e.returncode, e.output)
exceptions.IOError: ('Bad exit', ['flocker-changestate', 'nodes:\n  172.16.255.240: []\n  172.16.255.241: [mongodb-example-application]\nversion: 1\n', "applications:\n  mongodb-example-application:\n    image: clusterhq/mongodb\n    volume: {dataset_id: 565d4aee-9a88-43c4-ad5c-32ba0ea2941d, maximum_size: '104857600',\n      mountpoint: /data/db}\nversion: 1\n", '172.16.255.240:\n  applications: {}\n  used_ports: [22, 111, 48118, 48852]\n  version: 1\n172.16.255.241:\n  applications: {}\n  used_ports: [22, 111, 43085, 51276, 53274, 55954, 55967]\n  version: 1\n', '172.16.255.241'], 1, '')
                                  [FAIL]

```

**Comment by wallrj** [comment]
Yeah, I'm seeing this for all the acceptance tests

```
```

**Comment by wallrj** [comment]
Yeah, I'm seeing this for all the acceptance tests

```

```

**Comment by wallrj** [comment]
Sorry, didn't mean to close the PR.

**Comment by wallrj** [comment]
Same problem is showing up on buildbot:
 * http://build.clusterhq.com/builders/flocker%2Facceptance%2Fvagrant%2Ffedora-20/builds/352/steps/trial/logs/stdio

**Comment by wallrj** [comment]
I created some related issues:
 * https://clusterhq.atlassian.net/browse/FLOC-1273
 * https://clusterhq.atlassian.net/browse/FLOC-1272
 * https://clusterhq.atlassian.net/browse/FLOC-1274

**Pull request closed without merging**